packages/kit/src/core/config/index.js

```diff
@@ -91,7 +91,11 @@ function process_config(config, { cwd = process.cwd() } = {}) {
 		}
 	}
 
-	return validated;
+	if (validated.kit.adapter.config) {
+		return validated.kit.adapter.config(validated);
```

Comment:
users can supply custom adapters. validate should be called with the result of the adapter config hook